src/types.rs

```diff
@@ -0,0 +1,57 @@
+#[derive(PartialEq, Clone, Copy, Show)]
+pub struct Coordinate {
+    pub x: f64,
+    pub y: f64,
+}
+
+#[derive(PartialEq, Clone, Copy)]
+pub struct Point(pub Coordinate);
+
+#[derive(PartialEq, Clone)]
+pub struct MultiPoint(pub Vec<Point>);
+
+#[derive(PartialEq, Clone)]
+pub struct LineString(pub Vec<Point>);
+
+#[derive(PartialEq, Clone)]
+pub struct MultiLineString(pub Vec<LineString>);
+
+#[derive(PartialEq, Clone)]
+pub struct Polygon(pub LineString, pub Vec<LineString>);
```

Comment:
I think both are valid solutions, but for now, I vote to keep what you have just so we could get this merged as soon as possible. We can always change it later